.github/workflows/update-code-injection-models.yml

```diff
@@ -0,0 +1,73 @@
+name: 'Update code injection models'
+
+on:
+  # Run weekly (at arbitrary time)
+  schedule:
+    - cron: '31 7 * * 5'
+  workflow_dispatch:
+
+permissions: {}
+
+jobs:
+  update-models:
+    permissions:
+      contents: write
+      pull-requests: write
+    name: Update models
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4
+        with:
+          node-version: 22
+
+      # Use custom working dir for downloading npm packages and cloning the CodeQL repo
+      - name: Create working dir
+        run: |
+          mkdir codeql-models-working-dir
+
+      - name: Install dependencies
+        run: |
+          npm install yaml@2.7.1 --no-save
+        working-directory: codeql-models-working-dir
+
+      - name: Check out CodeQL
+        id: checkout_codeql
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
```

Comment:
Currently checks out the latest commit of the default branch. Might be nicer to check out the latest release tag instead for better stability. But maybe not worth it, and might not matter that much or at all assuming that (at least the `manual`) model files probably don't contain bugs which are only present on the latest default branch commits but not releases.

Reply:
I'm OK either way -- pulling from the default branch seems fine, since I can always reject PRs that look wrong.

(Getting the latest tag when there's no matching release is very annoying with GitHub's REST API -- you need to list the tags, then sort them by creation date, which requires addition roundtrips for each tag object. Not worth it, IMO :slightly_smiling_face:)